bundles/io/org.eclipse.smarthome.io.console/src/main/java/org/eclipse/smarthome/io/console/internal/extension/MetadataConsoleCommandExtension.java

```diff
@@ -87,14 +98,28 @@ public void execute(String[] args, Console console) {
         }
     }
 
-    private void listMetadata(Console console, String itemName, String namespace) {
-        MetadataKey key = new MetadataKey(namespace, itemName);
-        Metadata metadata = metadataRegistry.get(key);
-        if (metadata != null) {
-            console.println(metadata.toString());
+    private void listMetadata(Console console, String itemName, String namespace, boolean internal) {
+        if (itemName == null) {
+            metadataRegistry.stream().filter(m -> isInternal(m, internal)).map(Metadata::toString)
+                    .forEach(console::println);
+        } else if (namespace == null) {
+            metadataRegistry.stream().filter(MetadataPredicates.ofItem(itemName)).filter(m -> isInternal(m, internal))
+                    .map(Metadata::toString).forEach(console::println);
+        } else {
+            MetadataKey key = new MetadataKey(namespace, itemName);
+            if (metadataRegistry.isInternalNamespace(namespace) == internal) {
+                Metadata metadata = metadataRegistry.get(key);
+                if (metadata != null) {
+                    console.println(metadata.toString());
+                }
+            }
         }
     }
 
+    private boolean isInternal(@NonNull Metadata metadata, boolean internal) {
```

Comment:
We do not use `@NonNull` annotations instead we use `@NonNullByDefault` and `@Nullable`.